base/src/freetrm/eventstore/http/HttpReader.scala

```diff
@@ -0,0 +1,143 @@
+package freetrm.eventstore.http
+
+import java.net.URLEncoder
+
+import akka.NotUsed
+import akka.actor._
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.client.RequestBuilding._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{Accept, Authorization, BasicHttpCredentials}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import de.heikoseeberger.akkasse.MediaTypes.`text/event-stream`
+import de.heikoseeberger.akkasse.{EventStreamUnmarshalling, ServerSentEvent}
+import fommil.sjs.FamilyFormats._
+import freetrm.eventstore._
+import freetrm.eventstore.utils._
+import spray.json.JsonParser.ParsingException
+import spray.json._
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+class HttpReader(host: String, port: Int, username: String, cookie: String)
+                (implicit system: ActorSystem)
+  extends EventSourceReader with Log {
+
+  import EventStreamUnmarshalling._
+
+  implicit val mat = ActorMaterializer()
+  implicit val dispatcher = system.dispatcher
+
+  private val credentials = Authorization(BasicHttpCredentials(username, cookie))
+  private val streamHeader = Accept(`text/event-stream`)
+  private val textHeader = Accept(MediaTypes.`text/plain`)
+  private val headers = List(credentials, streamHeader, textHeader)
+
+  override def consume(topic: String,
+                       earliestOffsetToReadFrom: Long,
+                       lastOffsetToReadTo: Option[Long]): Source[Event, NotUsed] = {
+
+
+    val uri = s"/es/consume/${enc(topic)}/$earliestOffsetToReadFrom/" + lastOffsetToReadTo.map(_.toString).getOrElse("")
+    val req: Source[HttpResponse, NotUsed] = getRequest(uri)
+
+    consume[Event](req, response => s"Failed calling 'consume' for $topic: $response")
+  }
+  
+  override def consumeNotifications: Source[EventNotification, NotUsed] = {
+    val uri = "/es/consumeNotifications"
+    val req: Source[HttpResponse, NotUsed] = getRequest(uri)
+
+    consume[EventNotification](req, response => s"Failed calling 'consumeNotifications': $response")
+  }
+
+  override def surface(topic: String): Source[EventSourceEvent, NotUsed] = {
+    val uri = s"/es/surface/${enc(topic)}"
+    val req: Source[HttpResponse, NotUsed] = getRequest(uri)
+
+    consume[EventSourceEvent](req, response => s"Failed calling 'surface': $response")
+  }
+
+  private def consume[T](source: Source[HttpResponse, NotUsed], failMsg: String => String)
+                        (implicit writer: spray.json.JsonReader[T]) = {
+    source
+      .mapAsync(1) {
```

Comment:
reason for `mapAsync(1)` isn't obvious. I'd add a comment that although `mapAsync` guarantees order of future completion, `mapAsync(n > 1)` could cause a successful event to be streamed after a failure, whereas we would want to stop processing and (perhaps) retry.